nilearn/_utils/cache.py

```diff
@@ -132,7 +131,7 @@ def cache(func, memory, ref_memory_level=2, memory_level=1, **kwargs):
         returned.
     """
 
-    if ref_memory_level <= memory_level or memory is None:
+    if ref_memory_level > memory_level or memory is None:
```

Comment:
@AlexandreAbraham can you help me with this one?  The docstring says:
```
    ref_memory_level: int
        The reference memory_level used to determine if function call must
        be cached or not (if memory_level is larger than ref_memory_level
        the function is cached)
```

I believe this first check is for the not-caching condition, which means the code should be entered (according to the docstring) if `memory_level` is not larger than `ref_memory_level`.

Seems I should do  `ref_memory_level >= memory_level` on that reading...